jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -116,6 +120,23 @@ internal sealed class CodeScanSessionConfig(
         false -> "${getPayloadLimitInBytes() / TOTAL_BYTES_IN_KB}KB"
     }
 
+    open suspend fun getTotalProjectSizeInBytes(): Long {
+        var totalSize = 0L
+        try {
+            withTimeout(Duration.ofSeconds(TELEMETRY_TIMEOUT_IN_SECONDS)) {
+                VfsUtil.collectChildrenRecursively(projectRoot).filter {
+                    !it.isDirectory && !it.`is`((VFileProperty.SYMLINK)) && it.path.endsWith(sourceExt)
+                }.fold(0L) { acc, next ->
+                    totalSize = acc + next.length
+                    totalSize
+                }
+            }
+        } catch (e: TimeoutCancellationException) {
+            // Do nothing
+        }
+        return totalSize
+    }
+
```

Comment:
this can be a single expression
```suggestion
    open suspend fun getTotalProjectSizeInBytes(): Long =
        try {
            withTimeout(Duration.ofSeconds(TELEMETRY_TIMEOUT_IN_SECONDS)) {
                VfsUtil.collectChildrenRecursively(projectRoot).filter {
                    !it.isDirectory && !it.`is`((VFileProperty.SYMLINK)) && it.path.endsWith(sourceExt)
                }.fold(0L) { acc, next ->
                    acc + next.length
                }
            }
        } catch (e: TimeoutCancellationException) {
            -1
        }
```

Reply:
The idea is to return the partial total size calculated till the timeout.